src/sentry/static/sentry/app/components/avatarSettings.jsx

```diff
@@ -0,0 +1,515 @@
+                                      resizeDimensions.left * imgRatio,
+                                      resizeDimensions.top * imgRatio,
+                                      resizeDimensions.size * imgRatio,
+                                      resizeDimensions.size * imgRatio,
+                                      0, 0,
+                                      resizeDimensions.size * imgRatio,
+                                      resizeDimensions.size * imgRatio);
+    this.finishCrop();
+  },
+
+  finishCrop() {
+    let canvas = $(ReactDOM.findDOMNode(this.refs.canvas))[0];
+    this.props.updateDataUrlState({dataUrl: canvas.toDataURL()});
+  },
+
+  getImgSrc() {
+    let uuid = this.props.user.avatar.avatar_uuid;
+    let photoUrl = uuid && '/avatar/' + uuid + '/';
+    return this.props.savedDataUrl || this.state.objectURL || photoUrl;
+  },
+
+  onImgDrag(ev) {
+    ev.preventDefault();
+  },
+
+  renderImageCrop() {
+    let src = this.getImgSrc();
+    if (!src) {
+      return null;
+    }
+    let style = {
+      top: this.state.resizeDimensions.top,
+      left: this.state.resizeDimensions.left,
+      width: this.state.resizeDimensions.size,
+      height: this.state.resizeDimensions.size
+    };
+    return (
+      <div className="image-cropper">
+        <div className="crop-container" ref="cropContainer">
+          <div className="image-container">
+            <img className="preview" ref="image" src={src}
+                 onLoad={this.onLoad} onDragStart={this.onImgDrag}/>
+          </div>
+          <div className="cropper" style={style} onMouseDown={this.onMouseDown}>
+            <div onMouseDown={this.startResize.bind(this, 'nw')} className="resizer nw"></div>
+            <div onMouseDown={this.startResize.bind(this, 'ne')} className="resizer ne"></div>
+            <div onMouseDown={this.startResize.bind(this, 'se')} className="resizer se"></div>
+            <div onMouseDown={this.startResize.bind(this, 'sw')} className="resizer sw"></div>
+          </div>
+        </div>
+      </div>
+    );
+  },
+
+  uploadClick(ev) {
+    ev.preventDefault();
+    $(this.refs.file).click();
+  },
+
+  renderCanvas() {
+    if (!this.getImgSrc()) {
+      return null;
+    }
+    return (
+      <div className="canvas-container">
+        <canvas ref="canvas"></canvas>
+      </div>
+    );
+  },
+
+  render() {
+    let src = this.getImgSrc();
+    return (
+      <div>
+        {!src &&
+        <div className="image-well well blankslate">
+          <p><a onClick={this.uploadClick}><strong>Upload a photo</strong></a> to get started.</p>
+        </div>}
+        {this.renderImageCrop()}
+        {this.renderCanvas()}
+        <div className="form-group">
+          {src && <a onClick={this.uploadClick}>{t('Change Photo')}</a>}
+          <input ref="file" type="file" accept="image/*" onChange={this.onChange}/>
+        </div>
+      </div>
+    );
+  }
+});
+
+const AvatarRadio = React.createClass({
```

Comment:
i can split it up. it'd be better for testing to be able to import them individually anyway.